src/components/Cards/EventCard/EventCard.tsx

```diff
@@ -140,17 +139,40 @@ const EventCard: React.FC<EventCardProps> = (props) => {
       toast.error("User does not exist", { transition: Zoom });
     }
   };
+
+  const handleDeleteEvent = async () => {
```

Comment:
use the [Firebase RTDB remove function](https://firebase.google.com/docs/database/web/read-and-write#delete_data) instead of setting contents to null.